lib/src/client/connection.dart

```diff
@@ -101,13 +101,27 @@ class ClientConnection {
 
   @visibleForTesting
   Future<ClientTransportConnection> connectTransport() async {
-    final securityContext = options.credentials.securityContext;
+    final settings = options.http2.settings;
+
+    final connect = options.http2.connect;
+    if (connect != null) {
+      final streams = await connect(host, port);
+      if (_state == ConnectionState.shutdown) {
+        streams.outgoing.close();
```

Comment:
Shouldn't `incoming` also be closed ?

Reply:
I think `incoming` should be drained here. But it‘s the `producer`'s responsibility to close it.